src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -184,6 +185,48 @@ void UprobesUnwindingVisitor::visit(CallchainSamplePerfEvent* event) {
     return;
   }
 
+  // The leaf function is not guaranteed to have the frame pointer for all our targets. Though, we
+  // assume that $rbp remains untouched by the leaf functions, such that we can rely on
+  // perf_event_open to give us "almost" correct callstacks (the caller of the leaf function will be
+  // missing). We do a plausibility check for this assumption by checking if the callstack only
+  // contains executable code.
+  // TODO(b/187690455): As soon as we actually have frame pointers in all non-leaf functions, we can
+  //  report an unwinding error here. Till that point this check will always fail, because the
+  //  caller of __libc_start_main will be invalid since libc doesn't have frame-pointers. This
+  //  prevents us from testing the current implementation, which will have "almost" correct
+  //  callstack.
+  //  Note that, at this point in time, we don't want to actually throw the sample away, but rather
+  //  report it as "broken".
```

Comment:
Is this last part still relevant, now that in general we are no longer throwing samples away?

Reply:
Good point, removed it.